mailbox/store/src/test/java/org/apache/james/mailbox/store/quota/GenericMaxQuotaManagerTest.java

```diff
@@ -52,6 +55,14 @@ public void getMaxStorageShouldReturnEmptyWhenNoDefaultValue() throws Exception
         assertThat(maxQuotaManager.getMaxStorage(quotaRoot)).isEmpty();
     }
 
+    @Ignore("how can we link domain and quotaRoot ?")
```

Comment:
- This has to be done in `QuotaRootResolver`.

We need IMO to integrate that information to "QuotaRoot" (add it as a field in the object). Then `UserQuotaRootResolver` could simply retrieve it from the user object. (IE resolver becomes responsible to fix the domain, if any).

The problem then becomes "how do we retrieve the field when no resolver is used". IMO we add a field to QuotaRoot.quotaRoot for this.

Note: In non virtualHosting mode, there will be no domain. We have to ensure it still works.